contrib/IECoreUSD/test/IECoreUSD/USDSceneWriterTest.py

```diff
@@ -776,5 +776,51 @@ def testNameSanitisation( self ) :
 		scene = IECoreScene.SceneInterface.create( "/tmp/test.usda", IECore.IndexedIO.OpenMode.Read )
 		self.assertEqual( scene.childNames(), [ "_0", "wot_no_colons_", "fistFullOf___" ] )
 
+	def testQuatConversion( self ) :
+
+		# Imath and Gf quaternions do not share the same memory layout.
+		# Check that we account for that when writing.
+
+		root = IECoreScene.SceneInterface.create( "/tmp/test.usda", IECore.IndexedIO.OpenMode.Write )
+		child = root.createChild( "test" )
+		points = IECoreScene.PointsPrimitive( IECore.V3fVectorData() )
+		points["quatf"] = IECoreScene.PrimitiveVariable(
+			IECoreScene.PrimitiveVariable.Interpolation.Constant,
+			imath.Quatf( 0, imath.V3f( 1, 2, 3 ) )
+		)
+
+		child.writeObject( points, 0 )
+		del root, child
+
+		stage = pxr.Usd.Stage.Open( "/tmp/test.usda" )
+		primVarsAPI = pxr.UsdGeom.PrimvarsAPI( stage.GetPrimAtPath( "/test" ) )
+		quat = primVarsAPI.GetPrimvar( "quatf" ).Get( 0 )
+		self.assertEqual( quat, pxr.Gf.Quatf( 0, pxr.Gf.Vec3f( 1, 2, 3 ) ) )
+
+	def testRefCountAfterWrite( self ) :
+
+		root = IECoreScene.SceneInterface.create( "/tmp/test.usda", IECore.IndexedIO.OpenMode.Write )
+		child = root.createChild( "test" )
+
+		points = IECoreScene.PointsPrimitive( IECore.V3fVectorData() )
+		points["test1"] = IECoreScene.PrimitiveVariable(
+			IECoreScene.PrimitiveVariable.Interpolation.Constant,
+			IECore.IntVectorData( [ 1 ] )
+		)
+		points["test2"] = IECoreScene.PrimitiveVariable(
+			IECoreScene.PrimitiveVariable.Interpolation.Constant,
+			IECore.IntVectorData( [ 2 ] )
+		)
+
+		refCounts = { k : points[k].data.refCount() for k in points.keys() }
+
+		child.writeObject( points, 0 )
+		del root, child
+
+		self.assertEqual(
```

Comment:
I assume this would have failed previously, but definitely did not spot the change that fixes it... Can you elaborate?

Reply:
This wouldn't actually have failed previously. At one point in development I had an advanced version of the `toUSD()` conversion for `VectorData` which used a foreign data source to allow the USD `VtArray` to reference the memory held by the `VectorData` directly, with zero copying. But I discovered that USD wasn't always dropping the reference to the `VectorData` when I expected, and this test exposes that. I couldn't figure out the issue, but left the test in as a safety net for the next time I attempt this.

Reply:
Gotcha, thanks for the explanation